docs/source/reference/types.rst

```diff
@@ -54,3 +54,134 @@ mutating vector components will be added in a future release.
     v1 = float32x2(1.0, 1.0)
     v2 = float32x2(1.0, -1.0)
     dotprod = v1.x * v2.x + v1.y * v2.y
+
+
+Narrow Data Types
+~~~~~~~~~~~~~~~~~
+
+Bfloat16
+--------
+
+.. note::
+
+    Bfloat16 is only supported with CUDA version 12.0+, and only supported on
+    device compute capability with 8.0+.
+
+To determine whether Numba supports compiling code that uses the ``bfloat16``
+type in the current configuration, use:
+
+.. function:: numba.cuda.is_bfloat16_supported()
+
+    Returns ``True`` if current device supports bfloat16. ``False`` otherwise.
+
+Data Movement and Casts
+***********************
+
+Construction a single instance of bfloat16 object takes 1 API:
+
+.. function:: numba.cuda.bf16.bfloat16(a)
+
+    Constructs a bfloat16 data from existing device scalar. Supported scalar types:
+    - float64
+    - float32
+    - int64
+    - int32
+    - uint64
+    - uint32
+    On cuda version 12.0+ machines, supports construction from ``float16``
+
+Conversely, ``bfloat16`` data can be cast back to existing native data type via
+``dtype(h)``, where ``dtype`` is one of the data types above (except float16),
+and ``X`` is a bfloat16 object.
+
+Arithmatics
+************
+
+A ``bfloat16`` data can be computed with another ``bfloat16`` data with many
+supported aritheatic operators. The list of supported operations:
+
+- Arithmatics (``+, -, *, /``)
+- Arithmatic assignment oeprators (``+=, -=, *=, /=``)
+- Logical operators (``==, !=, >, <, >=, <=``)
+- Unary arithmatics (``+, -``)
+
+Math Intrinsics
+***************
+
```

Comment:
In this section there's some inconsistency in argument naming - sometimes it's `h` and sometimes `a` - I imagine `h` might have been followed from a convention for `float16` docs somewhere (i.e. `h` for half). 

Would it be better to always use `a`, or perhaps `x` for a generic argument name, or `b` for a `bfloat16`?

Reply:
I think the inconsistency comes from my simple mimic of https://docs.nvidia.com/cuda/cuda-math-api/cuda_math_api/group__CUDA__MATH____BFLOAT16__FUNCTIONS.html, which also has some of the inconsistencies.

I kinda like the idea of `b` for `bfloat16`, if that's ok with you, let's just adopt this name convention.

Reply:
Done [709ecf9](https://github.com/NVIDIA/numba-cuda/pull/245/commits/709ecf95757564cbab2dad8995fe9f08d06fd81b)